zooniverse/front-end-monorepo
# lib-user: Refactor UseStats, MyGroups, and GroupStats with containers

## Package
- lib-user

## Describe your changes
- refactor UserStats, MyGroups, and GroupStats with containers
- containers have hooks, fetch data, return components
- **no code changes**, largely copy, pasting, and cutting between component and container, with improvements to prop definitions and defaults, but no refactoring or new features (intentionally, as a lot of changes as is)

## Why?
- I think makes for more easily readable code
- I think will make subsequent changes easier and cleaner
- I think will make stories and tests for main components easier (in subsequent PRs)
- I think will make returning sad path components easier (in subsequent PRs, i.e. don't have access, nothing here, etc.)

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX? n/a
- What user actions should my reviewer step through to review this PR?
  - logged in as an admin user:
    - load a user stats page => should look and work ok:
      - lib-user (yarn dev): https://local.zooniverse.org:8080/?users=markb-panoptes/stats
      - app-root (yarn start): https://local.zooniverse.org:3000/users/markbouslog/stats
    - load a user's my groups page => should load, but design not updated:
      - lib-user (yarn dev): https://local.zooniverse.org:8080/?users=markb-panoptes/groups
      - app-root (yarn start): https://local.zooniverse.org:3000/users/markbouslog/groups
    - load a group stats page => should load, but design not updated:
      - lib-user (yarn dev): https://local.zooniverse.org:8080/?groups=1326945
      - app-root (yarn start): https://local.zooniverse.org:3000/groups/1326945
- Which storybook stories should be reviewed? n/a
- Are there plans for follow up PR’s to further fix this bug or develop this feature? yes, see linked GitHub project

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General

## Refactoring
